Services/GEV/Reports/classes/class.catBasicReportGUI.php

```diff
@@ -244,64 +244,55 @@ protected function makeGroupKey($row) {
 		return $head.$tail;
 	}
 
-	protected function exportXLS() {
-		require_once "Services/Excel/classes/class.ilExcelUtils.php";
-		require_once "Services/Excel/classes/class.ilExcelWriterAdapter.php";
-		
-		$data = $this->getData();
+	/**
+	* provide xlsx version of report for download.
+	*/
+	protected function exportXLSX() {
+		require_once 'Services/ReportsRepository/classes/class.catXLSXWriter.php';
 
-		$adapter = new ilExcelWriterAdapter("Report.xls", true); 
-		$workbook = $adapter->getWorkbook();
-		$worksheet = $workbook->addWorksheet();
-		$worksheet->setLandscape();
+		$workbook = new catXLSXWriter();
```

Comment:
Hast du eine Idee, wie das noch besser gestaltet werden kann? Wir haben ja nun ein Interface zum ExcelWriter, der Code in dieser Methode sollte auf jeder Implementierung funktionieren.